### ISSUE TITLE
VSS issues on Windows 10

### ISSUE THREAD

**Issue by Hoffenbar** [restic/restic#3390]
Output of `restic version`
--------------------------

restic 0.12.0 compiled with go1.15.8 on windows/amd64

How did you run restic exactly?
-------------------------------

```
RESTIC_PASSWORD=...
RESTIC_REPOSITORY=f:/Backups/Restic

C:\WINDOWS\system32>restic backup --use-fs-snapshot c:\Users\matthiasu\usr\pgm\tcl
repository 1f924bf5 opened successfully, password is correct
using parent snapshot d572c1de
creating VSS snapshot for [c:\]
error: failed to create snapshot for [c:\]: VSS error: async operation pending


/c/Users/matthiasu/usr/pgm/tcl/usr/Tst/minimal_vanillatclsh/32bit/tmp/vanillawish-78555c39f0-win32.exe/TDK/lib/dbg_engine/appLaunch.tcl
/c/Users/matthiasu/usr/pgm/tcl/usr/Tst/minimal_vanillatclsh/32bit/tmp/vanillawish-78555c39f0-win32.exe/TDK/lib/dbg_engine/blend.pdx
...
```

What backend/server/service did you use to store the repository?
----------------------------------------------------------------

USB drive

Expected behavior
-----------------

VSS snapshot should be created without error (works on Windows Server 2016)

Actual behavior
---------------

**error: failed to create snapshot for [c:\]: VSS error: async operation pending**

Steps to reproduce the behavior
-------------------------------

Se operations shown above. Would be great if someone else could test it on Windows 10 Home. Exact version of Windows 10 is:

```
C:\WINDOWS\system32>ver

Microsoft Windows [Version 10.0.19042.964]
```

Do you have any idea what may have caused this?
-----------------------------------------------

Yes - VSS seems to be complex (like many things on windows behind the scenes....).

Do you have an idea how to solve the issue?
-------------------------------------------

No.
Other programs like `shadowtask64.exe` could create a VSS snapshot without errors on my systems - so probably there isn't something wrong with my system in general.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------

Yes, it's a great tool. VSS is one of the features keeping it apart from kopia, which is otherwise a strong competitor and I'm experimenting with it, too, along with Duplicati.

**Comment by DRON-666** [comment]
You can try [this PR](https://github.com/restic/restic/pull/3067) with `-o vss.timeout` option.

**Comment by Hoffenbar** [comment]
Simply specifying `-o vss.timeout` doesn't helped. How to specify a concrete *value*?

**Comment by DRON-666** [comment]
For 10-minute timeout you can use `-o vss.timeout=10m` or `-o vss.timeout=600s`.

**Comment by Hoffenbar** [comment]
Unfortunally, specifying `-o vss.timeout=600s` didn't help. The error is the same (and appears before the timeout expires).

**Comment by Hoffenbar** [comment]
Maybe it has something to do with this:
```
C:\temp>vssadmin list writers
vssadmin 1.1 - Verwaltungsbefehlszeilenprogramm des Volumeschattenkopie-Dienstes
(C) Copyright 2001-2013 Microsoft Corp.

Auf Reaktion wird gewartet.
Die Reaktion ist möglicherweise verspätet, falls eine Schattenkopie vorbereitet wird.

Verfassername: "Task Scheduler Writer"
   Verfasserkennung: {d61d61c8-d73a-4eee-8cdd-f6f9786b7124}
   Verfasserinstanzkennung: {1bddd48e-5052-49db-9b07-b96f96727e6b}
   Status: [1] Stabil
   Letzter Fehler: Kein Fehler

Verfassername: "VSS Metadata Store Writer"
   Verfasserkennung: {75dfb225-e2e4-4d39-9ac9-ffaff65ddf06}
   Verfasserinstanzkennung: {088e7a7d-09a8-4cc6-a609-ad90e75ddc93}
   Status: [1] Stabil
   Letzter Fehler: Kein Fehler

Verfassername: "Performance Counters Writer"
   Verfasserkennung: {0bada1de-01a9-4625-8278-69e735f39dd2}
   Verfasserinstanzkennung: {f0086dda-9efc-47c5-8eb6-a944c3d09381}
   Status: [1] Stabil
   Letzter Fehler: Kein Fehler

Verfassername: "System Writer"
   Verfasserkennung: {e8132975-6f93-4464-a53e-1050253ae220}
   Verfasserinstanzkennung: {6959c76f-67cc-4788-bd86-e119f449dab8}
   Status: [1] Stabil
   Letzter Fehler: Kein Fehler

Verfassername: "ASR Writer"
   Verfasserkennung: {be000cbe-11fe-4426-9c58-531aa6355fc4}
   Verfasserinstanzkennung: {153c439a-fbd9-428c-baa3-46afaea8ead3}
   Status: [10] Fehler
   Letzter Fehler: Zeitüberschreitung

Verfassername: "WMI Writer"
   Verfasserkennung: {a6ad56c2-b509-4e6c-bb19-49d8f43532f0}
   Verfasserinstanzkennung: {49d5bbb2-756d-4f52-97df-305df7308105}
   Status: [10] Fehler
   Letzter Fehler: Zeitüberschreitung

Verfassername: "MSSearch Service Writer"
   Verfasserkennung: {cd3f2362-8bef-46c7-9181-d62844cdc0b2}
   Verfasserinstanzkennung: {c44e3484-d2f5-433d-b4c2-6ff611463db8}
   Status: [10] Fehler
   Letzter Fehler: Zeitüberschreitung

Verfassername: "IIS Config Writer"
   Verfasserkennung: {2a40fd15-dfca-4aa8-a654-1f8c654603f6}
   Verfasserinstanzkennung: {29fe7096-f6f1-41c4-98fd-cedce8fcf861}
   Status: [10] Fehler
   Letzter Fehler: Zeitüberschreitung

Verfassername: "Shadow Copy Optimization Writer"
   Verfasserkennung: {4dc3bdd4-ab48-4d07-adb0-3bee2926fd7f}
   Verfasserinstanzkennung: {c04a953d-aeb0-4495-b3b7-1750a3d2e6bf}
   Status: [10] Fehler
   Letzter Fehler: Zeitüberschreitung

Verfassername: "Registry Writer"
   Verfasserkennung: {afbab4a2-367d-4d15-a586-71dbb18f8485}
   Verfasserinstanzkennung: {b6c52a7c-d0d5-45e2-b353-f2ce932ed912}
   Status: [10] Fehler
   Letzter Fehler: Zeitüberschreitung

Verfassername: "COM+ REGDB Writer"
   Verfasserkennung: {542da469-d3e1-473c-9f4f-7847f01fc64f}
   Verfasserinstanzkennung: {28c662a2-8e27-4210-bdd2-7ba51935f1a7}
   Status: [10] Fehler
   Letzter Fehler: Zeitüberschreitung
```

**Comment by Hoffenbar** [comment]
In the windows eventlog, I see this:
```
Volumeschattenkopie-Dienstfehler: Beim Abfragen nach der Schnittstelle "IVssWriterCallback" ist ein unerwarteter Fehler aufgetreten. hr = 0x80070005, Zugriff verweigert
. Die Ursache hierfür ist oft eine falsche Sicherheitseinstellung im Schreib- oder Anfrageprozess. 

Vorgang:
   Generatordaten werden gesammelt

Kontext:
   Generatorklassen-ID: {e8132975-6f93-4464-a53e-1050253ae220}
   Generatorname: System Writer
   Generatorinstanz-ID: {6959c76f-67cc-4788-bd86-e119f449dab8}
```

**Comment by k2s** [comment]
I found this [article](https://www.yourdigitalmind.com/tutorials/how-to-fix-volume-shadow-copy-service-error-unexpected-error-querying-for-the-ivsswritercallback-interface/) and already Solution 1, point 2 helped me to resolve this problem.

**Comment by JsBergbau** [comment]
I had this error operation pending sometimes because restic backup and Windows automatic shadow copies tried to make a snapshot at the same time. I changed restic interval so restic can make the snapshot when no other process creates one. 

regarding Duplicati: Try to restore files without the database. I tried to restore about 1,5 TB and it took over one week. Thats why Duplicati is no option for me. 

Regarding Kopia: I only see compression as really feature that would make it worth for me a look.
